### PR TITLE
fix: add test instructions for wrangler init

### PR DESCRIPTION
**Pull request by rozenmd**

### CODE REVIEW

File: packages/wrangler/templates/test-new-worker.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { unstable_dev } from "wrangler";
+const { unstable_dev } = require("wrangler");
```
Comment (penalosa) [inline packages/wrangler/templates/test-new-worker.js:1]: Why can't this be an ESM import?

File: packages/wrangler/src/init.ts [hunk 2]
```diff
@@ -415,6 +415,9 @@ export async function initHandler(args: ArgumentsCamelCase<InitArgs>) {
 					isNamedWorker ? `cd ${args.name} && ` : ""
 				}npm start\``
 			);
+			if (isAddingTests && !packageJsonContent.scripts?.test) {
```
Comment (penalosa) [inline packages/wrangler/src/init.ts:418]: Not sure I understand this—is this saying to only print the instruction line if there's no test script?

**Review by penalosa — CHANGES_REQUESTED** [review]

File: packages/wrangler/templates/test-new-worker.js (hunk 1 above)
Comment (rozenmd) [reply packages/wrangler/templates/test-new-worker.js:1]: Setting up jest for ESM can be a follow-up task

File: packages/wrangler/src/init.ts (hunk 2 above)
Comment (rozenmd) [reply packages/wrangler/src/init.ts]: I've extracted these out into their own variable to be clearer, cheers

File: packages/wrangler/templates/test-new-worker.js (hunk 1 above)
Comment (penalosa) [reply packages/wrangler/templates/test-new-worker.js:1]: Does naming it with mjs instead of js work?

**Review by penalosa — APPROVED** [review]

File: packages/wrangler/templates/test-new-worker.js (hunk 1 above)
Comment (rozenmd) [reply packages/wrangler/templates/test-new-worker.js:1]: It's not trivial to set-up: https://stackoverflow.com/a/69059786/5175932

Comment (rozenmd) [reply packages/wrangler/templates/test-new-worker.js:1]: Bare minimal setup requires adding `"type": "module",` to package.json, installing `cross-env`, and setting the test script to `cross-env NODE_OPTIONS=--experimental-vm-modules jest`

**Pull request merged**